extensions/vscode/src/views/homeView.ts

```diff
@@ -1199,6 +1271,13 @@ export class HomeViewProvider implements WebviewViewProvider, Disposable {
     }
   };
 
+  public initiatePublish(target: PublishProcessParams) {
+    this.onDeployMsg({
+      kind: WebviewToHostMessageType.DEPLOY,
+      content: target,
+    });
+  }
```

Comment:
This feels a bit strange to me. It isn't sending a message, but is passing it just because `onDeployMsg` is the thing that actually starts the deploy with `deployProject`.

Maybe we should split that logic out of there, and make `onDeployMsg` call the new function, and remove this middle man.